src/overlays/actors/ovl_Bg_Mori_Idomizu/z_bg_mori_idomizu.c

```diff
@@ -20,23 +35,156 @@ const ActorInit Bg_Mori_Idomizu_InitVars = {
     NULL,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mori_Idomizu/func_808A4640.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mori_Idomizu/func_808A4648.s")
+void BgMoriIdomizu_SetActionFunction(BgMoriIdomizu* this, BgMoriIdomizuActionFunc actionFunc) {
+    this->actionFunc = actionFunc;
+}
+
+void BgMoriIdomizu_SetWaterLevel(GlobalContext* globalCtx, s16 waterLevel) {
+    WaterBox* waterBox;
+
+    waterBox = globalCtx->colCtx.stat.colHeader->waterBoxes;
+    waterBox[2].unk_02 = waterLevel;
+    waterBox[3].unk_02 = waterLevel;
+    waterBox[4].unk_02 = waterLevel;
+}
+
+void BgMoriIdomizu_Init(Actor* thisx, GlobalContext* globalCtx) {
+    GlobalContext* globalCtx2 = globalCtx;
+    BgMoriIdomizu* this = THIS;
+
+    if (alreadyLoaded != 0) {
+        Actor_Kill(&this->actor);
+        return;
+    }
+    this->actor.scale.x = 1.1f;
+    this->actor.scale.y = 1.0f;
+    this->actor.scale.z = 1.0f;
+    this->actor.posRot.pos.x = 119.0f;
+    this->actor.posRot.pos.z = -1820.0f;
+    this->switchFlag = Flags_GetSwitch(globalCtx2, this->actor.params & 0x3F);
+    if (this->switchFlag != 0) {
+        this->actor.posRot.pos.y = -282.0f;
+        BgMoriIdomizu_SetWaterLevel(globalCtx2, -282);
+    } else {
+        this->actor.posRot.pos.y = 184.0f;
+        BgMoriIdomizu_SetWaterLevel(globalCtx2, 184);
+    }
+    this->objBankIndex = Object_GetIndex(&globalCtx2->objectCtx, OBJECT_MORI_TEX);
+    if (this->objBankIndex < 0) {
+        Actor_Kill(&this->actor);
+        // Bank danger!
+        osSyncPrintf("Error : バンク危険！(arg_data 0x%04x)(%s %d)\n", this->actor.params, "../z_bg_mori_idomizu.c",
+                     202);
+        return;
+    }
+    BgMoriIdomizu_SetupObjectCheck(this);
+    alreadyLoaded = 1;
+    this->isLoaded = 1;
+    this->actor.room = -1;
+    // Forest Temple well water
+    osSyncPrintf("(森の神殿 井戸水)(arg_data 0x%04x)\n", this->actor.params);
+}
+
+void BgMoriIdomizu_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    GlobalContext* globalCtx2 = globalCtx;
+    BgMoriIdomizu* this = THIS;
+
+    if (this->isLoaded) {
+        alreadyLoaded = 0;
+    }
+}
+
+void BgMoriIdomizu_SetupObjectCheck(BgMoriIdomizu* this) {
+    BgMoriIdomizu_SetActionFunction(this, BgMoriIdomizu_ObjectCheck);
+}
+
+void BgMoriIdomizu_ObjectCheck(BgMoriIdomizu* this, GlobalContext* globalCtx) {
+    if (Object_IsLoaded(&globalCtx->objectCtx, this->objBankIndex)) {
+        BgMoriIdomizu_SetupAction(this);
+        this->actor.draw = BgMoriIdomizu_Draw;
+    }
+}
+
+void BgMoriIdomizu_SetupAction(BgMoriIdomizu* this) {
+    BgMoriIdomizu_SetActionFunction(this, &BgMoriIdomizu_Action);
+}
+
+void BgMoriIdomizu_Action(BgMoriIdomizu* this, GlobalContext* globalCtx) {
+    s8 roomNum;
+    Actor* thisx = &this->actor;
+    s32 newSwitchFlag;
+
+    roomNum = globalCtx->roomCtx.curRoom.num;
+    newSwitchFlag = Flags_GetSwitch(globalCtx, thisx->params & 0x3F);
+    if (newSwitchFlag != 0) {
```

Comment:
```suggestion
    switchFlagSet = Flags_GetSwitch(globalCtx, thisx->params & 0x3F);
    if (switchFlagSet) {
```
cf comment in `Init` near `Flags_GetSwitch` (also, use as boolean)